question-generation/nlp.py

```diff
@@ -0,0 +1,42 @@
+from transformers import pipeline
+
+
+def interview_ai(sequence: str, num_questions: int) -> "list[str]":
+    """Generate questions based on a given sequence of context using Interview AI.
+
+    Args:
+        sequence: A string representing the context from which to generate questions.
+        num_questions: The number of questions to generate.
+
+    Returns:
+        A list of generated questions as strings.
+    """
+
+    model_id = "hyechanjun/interview-question-remake"
+    pipe = pipeline("text2text-generation", model=model_id, device=0)
+
+    outputs = pipe(
+        sequence,
+        max_length=64,
+        min_length=9,
+        num_beams=4,
+        num_return_sequences=num_questions,
```

Comment:
What if `num_questions` > `num_beams * num_beam_groups` (or whatever the total size is, not sure I understand how beam grouping is specified)

Reply:
Thank you for catching that! According to [this blog](https://huggingface.co/blog/how-to-generate), `num_return_sequences` must be less than or equal to `num_beams`. Configuring `num_beam_groups` helps with balancing the diversity of beam search. Huggingface documentation references [this paper](https://arxiv.org/abs/1610.02424). 

I don't know if setting `num_beams` = `num_beam_groups` necessarily results in the most diverse sequences, and this is something that we will have to experiment with.

For our purposes we can go in several directions:

1. Replace it with `min(4, num_questions)`, so that it won't run into `ValueError`.
2. Set `num_beams`, `num_beam_groups`, and `num_return_sequences` to `num_questions`.
3. Let global variables determine the value of these parameters. We can configure these values later, before running user studies. This will depend on how [this comment](https://github.com/AIToolsLab/questions/pull/13/files/fb638124bcc2a9c03d3ac2c5592dc761c3f423ea#r1245892538) is resolved, and it would potentially help resolve [this comment](https://github.com/AIToolsLab/questions/pull/13/files/fb638124bcc2a9c03d3ac2c5592dc761c3f423ea#r1245891658).
4. Other suggestions?

Reply:
https://huggingface.co/docs/transformers/v4.30.0/en/main_classes/text_generation#transformers.GenerationConfig.num_beam_groups clarifies that beam groups *divides* `num_beams` into groups. So `num_beams` still controls the total number of results.

I'd definitely want to experiment with setting this to a larger value, at least twice `num_beams`. But let's take that sort of thing into a separate PR.